MSAL/src/native_auth/controllers/sign_up/MSALNativeAuthSignUpController.swift

```diff
@@ -246,7 +257,8 @@ final class MSALNativeAuthSignUpController: MSALNativeAuthBaseController, MSALNa
         context: MSIDRequestContext
     ) -> SignUpStartControllerResponse {
         switch result {
-        case .codeRequired(let sentTo, let challengeType, let codeLength, let continuationToken):
+        case .codeRequired(let sentTo, let challengeType, let codeLength, let continuationToken, let correlationId):
+            let context = updateContext(context, with: correlationId)
             MSALLogger.log(level: .info, context: context, format: "Successful signup/challenge request")
             return SignUpStartControllerResponse(
                 .codeRequired(
```

Comment:
at line 268 we use the correlationId from the context instead of the one received from the API. Is this behaviour expected?